pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -908,11 +905,16 @@ func (c *VSphereController) deleteConfigMapIfExists(ctx context.Context, name, n
 }
 
 func ConvertToPlatformSpec(config *vclib.VSphereConfig, infra *ocpv1.Infrastructure) {
+	if infra.Spec.PlatformSpec.VSphere == nil {
+		// Add an empty vSphere spec to infra created in OCP 4.10 and earlier.
+		// It gets populated from the `config` object below.
+		infra.Spec.PlatformSpec.VSphere = &ocpv1.VSpherePlatformSpec{}
+	}
 	vSphereSpec := infra.Spec.PlatformSpec.VSphere
```

Comment:
On second thoughts - do we need some unit tests for this? It should be simple to write some unit tests that fixes this.

Reply:
Unit test added. I needed to add nil check to the test code itself.